src/parquet/schema.cc

```diff
@@ -620,6 +620,15 @@ const ColumnDescriptor* SchemaDescriptor::Column(int i) const {
   return &leaves_[i];
 }
 
+int SchemaDescriptor::ColumnIndex(const NodePtr& node) const {
```

Comment:
@wesm do you have a solution for this situation? shared_ptr as keys to an `unordered_map` indeed seems complicated. But I also think O(n) must be avoided.